pandas/core/frame.py

```diff
@@ -8764,11 +8768,22 @@ def update(
         if not isinstance(other, DataFrame):
             other = DataFrame(other)
 
-        other = other.reindex(self.index)
+        if other.index.has_duplicates:
+            raise ValueError("Update not allowed with duplicate indexes on other.")
+
+        rows = other.index.intersection(self.index)
+        if rows.empty:
+            raise ValueError(
+                "Can't update dataframe when other has no index in common with "
```

Comment:
No big deal, but in the previous error message you say "update not allowed" and here "can't update dataframe" to mean the same thing. Not sure how we usually write error messages, but I think it'd be better to be consistent.